frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -1251,9 +1252,24 @@ class Worksheet extends React.Component {
         this.setState({ BulkBundleDialog: deleteWorksheetDialog });
     }
 
+    onShowNewUpload = (e) => {
```

Comment:
Can you rename this to something like `showUploadMenu` (so that it doesn't sound too much like `onShowNewRun` or `onShowNewText`, which do very different things)? (and before #1945, we used to have a variable called `showNewUpload` as well)